exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -179,27 +190,29 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, extensionConfig *runti
 	return ctrl.Result{}, nil
 }
 
-// secretToExtensionConfig maps a secret to ExtensionConfigs with the corresponding InjectCAFromSecretAnnotation
-// to reconcile them on updates of the secrets.
-func (r *Reconciler) secretToExtensionConfig(ctx context.Context, secret client.Object) []reconcile.Request {
-	result := []ctrl.Request{}
-
-	extensionConfigs := runtimev1.ExtensionConfigList{}
-	indexKey := secret.GetNamespace() + "/" + secret.GetName()
+// secretToExtensionConfigFunc returns a func which maps a secret to ExtensionConfigs with the corresponding
+// InjectCAFromSecretAnnotation to reconcile them on updates of the secrets.
+func secretToExtensionConfigFunc(ctrlClient client.Client) handler.TypedMapFunc[*metav1.PartialObjectMetadata] {
```

Comment:
I was playing a bit around with this. I think this change can be entirely reverted. The only change we need is this
```go
func (r *Reconciler) secretToExtensionConfig(ctx context.Context, secret *metav1.PartialObjectMetadata) []reconcile.Request {
```

This can then be passed in above like this:
```go
			handler.TypedEnqueueRequestsFromMapFunc(
				r.secretToExtensionConfig,
			),
```

Reply:
I think the generic func is only needed in the CRS controller, because there we actually use it with different types